### PR TITLE
bug fix: error when both page_by and group_by are used.

### PR DESCRIPTION
**Pull request by fb-elong**
Fix bug identified in #168. 

The code now work as expected as in the original issue. 

image

### CODE REVIEW

File: R/as_rtf_pageby.R [hunk 1]
```diff
@@ -139,10 +139,18 @@ as_rtf_pageby <- function(tbl) {
 
   ## Remove repeated records if group_by is not null
   if (!is.null(group_by)) {
-    cell_tbl <- rtf_group_by_enhance(cell_tbl,
-      group_by = group_by,
+    tmp <- cell_tbl
+    tmp$`.pageby` <- pageby$id
+    tmp$`.order` <- 1:nrow(tmp)
```
Comment (nanxstats) [inline R/as_rtf_pageby.R:144]: `1:nrow(tmp)` is likely to be wrong in the empty edge case. Use `seq_len(nrow(tmp))` instead.

File: R/as_rtf_pageby.R [hunk 2]
```diff
@@ -139,10 +139,18 @@ as_rtf_pageby <- function(tbl) {
...
+    tmp <- rtf_group_by_enhance(tmp,
+      group_by = c(".pageby", group_by),
       page_index = subset(page_dict, !pageby)$page
     )
+
+    stopifnot(all(1:nrow(tmp) == tmp$`.order`))
```
Comment (nanxstats) [inline R/as_rtf_pageby.R:150]: Same issue on `1:nrow(tmp)`

File: R/as_rtf_pageby.R [hunk 3]
```diff
@@ -139,10 +139,18 @@ as_rtf_pageby <- function(tbl) {
...
+    tmp <- tmp[, !names(tmp) %in% c(".pageby", ".order")]
```
Comment (nanxstats) [inline R/as_rtf_pageby.R:151]: To preserve the original dimensionality, use `drop = FALSE`. https://adv-r.hadley.nz/subsetting.html#simplify-preserve

**Review by nanxstats — CHANGES_REQUESTED** [review]
I added a few comments on common issues.

File: R/as_rtf_pageby.R (hunk 3 above)
Comment (fb-elong) [reply R/as_rtf_pageby.R]: updated

File: R/as_rtf_pageby.R (hunk 2 above)
Comment (fb-elong) [reply R/as_rtf_pageby.R]: updated

**Review by nanxstats — APPROVED** [review]
Looking good! Thanks for the fix and the update.

**Pull request merged**